### PR TITLE
Fixed a bunch of antag stuff

### LINKED ISSUE #676: tator AIs don't get their tator law or malf tab at roundstart
1. get traitor ai
2. no traitor law
3. no objectives
4. no malf tab
5. must be retraitored to fix

### PR DESCRIPTION
**Pull request by Cruix**
Fixed traitor AIs not spawning correctly
Added secret to force traitor AI
Fixed assassination targets not being crewmembers (stolen from https://github.com/tgstation/tgstation/pull/19937/files)
Consolidated create-antagonists code

Fixes #676

### CODE REVIEW

File: code/game/gamemodes/game_mode.dm [hunk 1]
```diff
@@ -555,12 +563,31 @@
...
-/*
-/datum/game_mode/proc/check_player_role_pref(var/role, var/mob/new_player/player)
-	if(player.preferences.be_special & role)
-		return 1
-	return 0
-*/
+/datum/game_mode/proc/get_playing_crewmembers_for_role(role, restricted_jobs_for, on_station = 1)
+	var/list/mob/living/carbon/human/candidates = list()
+	var/list/crewmembers = list()
+	for(var/V in data_core.locked)
```
Comment (X-TheDark) [inline code/game/gamemodes/game_mode.dm:569]: I'm curious, why are you using `data_core.locked`, instead of, say `player_list`?

Comment (Cruix) [reply code/game/gamemodes/game_mode.dm:569]: Because player_list includes ashwalkers, free golems, etc.

Comment (Alek2ander) [reply code/game/gamemodes/game_mode.dm:569]: You can look at the player list and check special role, pretty sure all non-station beings have one

Comment (Cruix) [reply code/game/gamemodes/game_mode.dm:569]: No, none of the lavaland roles I have looked at have a special role.

**Comment by timkoster1** [comment]
Changelog!!

**Pull request merged**